src/js/stores/MesosStateStore.js

```diff
@@ -128,20 +128,15 @@ var MesosStateStore = Store.createStore({
     let services = this.get('lastMesosState').frameworks;
     let foundTask = null;
 
-    for (let serviceIndex in services) {
-      let service = services[serviceIndex];
+    services.some(function (service) {
       let tasks = service.tasks.concat(service.completed_tasks);
 
-      for (let taskIndex in tasks) {
-        let task = tasks[taskIndex];
-        let equalTask = task.id === taskID;
+      foundTask = tasks.find(function (task) {
+        return task.id === taskID;
+      });
 
-        if (equalTask) {
-          foundTask = task;
-          break;
-        }
-      }
-    }
+      return foundTask;
+    });
```

Comment:
Would be nice to make it more efficient instead rather than looping through all the services:
```

foo:
for (var i = 0; i < 100000; i++) {
  for (var j = 0; j < 9999999; j++) {
    if (j === 99) {
      break foo;
    }
  }
}
console.log(i, j);
```